components/emoji_picker/utils/index.ts

```diff
@@ -70,27 +94,31 @@ export function getFilteredEmojis(allEmojis: Record<string, Emoji>, filter: stri
 
     const filteredEmojis = [...sortedRecentEmojis, ...sortedFiltertedEmojisMinusRecent];
 
-    return filteredEmojis;
+    const filteredEmojisUserSkinTone = filteredEmojis.filter((emoji) => {
+        return emojiMatchesSkin(emoji, userSkinTone);
+    });
+
+    return filteredEmojisUserSkinTone;
 }
 
 // Note : This function is not an idea implementation, a more better and efficeint way to do this come when we make changes to emoji json.
-function convertEmojiToUserSkinTone(emoji: SystemEmoji, emojiSkin: string, userSkinTone: string) {
+function convertEmojiSkinTone(emoji: SystemEmoji, emojiSkin: string, skinTone: string) {
```

Comment:
Some of these changes are going to conflict with https://github.com/mattermost/mattermost-webapp/pull/10321 since I moved some of this around and made some similar changes as well. I think that's fine though. I'll wait to merge my PR until after this one is in